packages/shared_preferences/shared_preferences_ios/ios/Classes/FLTSharedPreferencesPlugin.m

```diff
@@ -80,4 +21,71 @@ + (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar> *)registrar {
   return filteredPrefs;
 }
 
+@interface FLTSharedPreferencesPlugin () <SharedPreferencesApi>
+@end
+
+@implementation FLTSharedPreferencesPlugin
+
++ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar> *)registrar {
+  FLTSharedPreferencesPlugin *plugin = [[FLTSharedPreferencesPlugin alloc] init];
+  SharedPreferencesApiSetup(registrar.messenger, plugin);
+}
+
+- (nullable NSNumber *)clearWithError:(FlutterError *_Nullable __autoreleasing *_Nonnull)error {
+  NSUserDefaults *defaults = [NSUserDefaults standardUserDefaults];
+  for (NSString *key in getAllPrefs()) {
+    [defaults removeObjectForKey:key];
+  }
+  return @YES;
+}
+
+- (nullable NSDictionary<NSString *, id> *)getAllWithError:
+    (FlutterError *_Nullable __autoreleasing *_Nonnull)error {
+  return getAllPrefs();
+}
+
+- (nullable NSNumber *)removeKey:(nonnull NSString *)key
+                           error:(FlutterError *_Nullable __autoreleasing *_Nonnull)error {
+  [[NSUserDefaults standardUserDefaults] removeObjectForKey:key];
+  return @YES;
+}
+
+- (nullable NSNumber *)setBoolKey:(nonnull NSString *)key
+                            value:(nonnull NSNumber *)value
+                            error:(FlutterError *_Nullable __autoreleasing *_Nonnull)error {
+  [[NSUserDefaults standardUserDefaults] setBool:value.boolValue forKey:key];
+  return @YES;
+}
+
+- (nullable NSNumber *)setDoubleKey:(nonnull NSString *)key
+                              value:(nonnull NSNumber *)value
+                              error:(FlutterError *_Nullable __autoreleasing *_Nonnull)error {
+  [[NSUserDefaults standardUserDefaults] setDouble:value.doubleValue forKey:key];
+  return @YES;
+}
+
+- (nullable NSNumber *)setIntKey:(nonnull NSString *)key
+                           value:(nonnull NSNumber *)value
+                           error:(FlutterError *_Nullable __autoreleasing *_Nonnull)error {
+  // int type in Dart can come to native side in a variety of forms
+  // It is best to store it as is and send it back when needed.
+  // Platform channel will handle the conversion.
+  [[NSUserDefaults standardUserDefaults] setValue:value forKey:key];
+  return @YES;
+}
+
+- (nullable NSNumber *)setStringKey:(nonnull NSString *)key
+                              value:(nonnull NSString *)value
+                              error:(FlutterError *_Nullable __autoreleasing *_Nonnull)error {
+  [[NSUserDefaults standardUserDefaults] setValue:value forKey:key];
+  return @YES;
+}
+
+- (nullable NSNumber *)setStringListKey:(nonnull NSString *)key
+                                  value:(nonnull NSArray<NSString *> *)value
+                                  error:(FlutterError *_Nullable __autoreleasing *_Nonnull)error {
+  [[NSUserDefaults standardUserDefaults] setValue:value forKey:key];
+  return @YES;
+}
+
```

Comment:
I don't think lines of code is the correct way to measure the change because of objc formatting and the need to distinguish between compile time verified lines and runtime verified lines.  Here's instead how the changes should be measured:
1) Every method went from ~6 operations to ~2.
1) Every method removed 2 casts
1) Conformance to the API is enforced at compile time
1) Error handling has been made available to the objc handlers

Reply:
Another thing to note is that xcode generated most of this code so it didn't feel heavy weight when writing it.  Once you implement a protocol xcode offers to fill it in.